### ISSUE TITLE
make thousands of vagrant "box"es importable

### ISSUE THREAD

**Issue by OpenNebulaProject** [OpenNebula/one#659]
---

Author Name: **Florian Heigl** (Florian Heigl)
Original Redmine Issue: 2111, https://dev.opennebula.org/issues/2111
Original Date: 2013-06-01

---

Hi,

I want to propose a feature that allows importing premade virtual machine images that have been prepared using vagrant.
What would it do:

* allows users to build opennebula / vagrant dual use VMs on their silly laptop and upload them to the ONE IaaS
* allows users to use the Sunstone interface to import a vagrant VM as a new template simply via it's URL

What I think why this is possible / interesting

* should work like vagrant does it, that means by using a certain URL
* concepts are similar, base image and instantiation
* allow local importing of vagrant box images
* increase the marketplace volume with tested images
* save on work for providing images in the marketplace (!!!) -> meaning more reliable and current.
* vagrant per se is "cool" for many people, especially those that can't pull full PXE setups out of their bag.
* help larger dev teams use the "full cloud power"
* need to be aware of security implications of vagrant boxes that everyone keeps ignoring (thats why "tested images" above)
* ...oracle vm has a kind of staging where you can "clear" a image to run, or PULL it. 
* ...might be smart to have something: once the vagrant community starts looking into how many of their images are w/rootkits already, ONE could win by being able to revoke a certain images access rights etc. You know: All the cool stuff virtual infrastructure allows.

notes:
* virtualbox dependency can probably be solved - i.e. automated testing on diff hypervisors. compatible HV selection is already solved in ONE

I've kept this shorter than I wanted, but I already spent 8 days pondering the text.
So please just use your imagination to see how many options this gives, and feel free to "help" by editing.

**Comment by OpenNebulaProject** [comment]
---

Original Redmine Comment
Author Name: **Ruben S. Montero** (@rsmontero)
Original Date: 2013-06-09T18:06:13Z

---

Would it make sense to develop an OpenNebula-Vagrant provider? As described here for AWS....

http://docs.vagrantup.com/v2/plugins/providers.html

**Comment by OpenNebulaProject** [comment]
---

Original Redmine Comment
Author Name: **Florian Heigl** (Florian Heigl)
Original Date: 2013-06-13T21:16:52Z

---

I think that makes a lot sense (but is another topic)
Quick to use interfaces could give Opennebula more reputation than any professional feature (we love)

Amazon might be a little pissed about the lost revenue. :)

**Comment by OpenNebulaProject** [comment]
---

Original Redmine Comment
Author Name: **Valentin Bud** (Valentin Bud)
Original Date: 2014-10-23T12:35:28Z

---

For whom it may concern there is an OpenNebula Vagrant provider 
out there in the wild. It doesn't have all the features provided by 
Virtualbox to say, but it works to start, stop VMs on an OpenNebula
deployment.

It require rOCCI.

https://github.com/eucher/opennebula-provider

Best,
Valentin

**Comment by OpenNebulaProject** [comment]
---

Original Redmine Comment
Author Name: **Florian Heigl** (Florian Heigl)
Original Date: 2016-04-15T23:05:41Z

---

Hi, 

I came back to this. The opennebula provider can start templates in OpenNebula, nothing else.
There's also a branch without rOCCI requirement and one that adds suspend/resume, which is nice.


But anyone who sits down with that on the one side and a normal vagrant setup on the other side will notice that the core functionality of vagrant includes retrieving images from ((often) external) sources.

A real solution would need to fetch the 'box' images and wrap them in a template. There's not really anything technical in the way, except that it's more than a day's work and requires strong ONE dev background.

What puzzles me is why this rotted away in the backlog with surely not more than 30 minutes of attention, for something that could've gained a very very large additional user base. Now that the dev crowd moved to docker those are all lost and won't come back. So basically you specifically opted to shrink your user base.

Anyway. Once I got this working in general I'll try to at least create some hook that is able to fetch unpack things and report back.
